🐛 [Bug]: EnvVar middleware parsing error

### Bug Description

```go
func TestEnvVarHandlerSpecialValue(t *testing.T) {
	os.Setenv("testEnvKey", "testBase64:TQ==")
	defer os.Unsetenv("testEnvKey")

	app := fiber.New()
	app.Use("/envvars", New())

	req, _ := http.NewRequest("GET", "http://localhost/envvars", nil)
	resp, err := app.Test(req)
	utils.AssertEqual(t, nil, err)

	respBody, err := ioutil.ReadAll(resp.Body)
	utils.AssertEqual(t, nil, err)

	var envVars EnvVar
	utils.AssertEqual(t, nil, json.Unmarshal(respBody, &envVars))
	val := envVars.Vars["testEnvKey"]
	utils.AssertEqual(t, "testBase64:TQ==", val)
}
```

```go
=== RUN   TestEnvVarHandlerSpecialValue
    envvar_test.go:124: 
        Test:       TestEnvVarHandlerSpecialValue
        Trace:      envvar_test.go:124
        Expect:     testBase64:TQ==     (string)
        Result:     testBase64:TQ       (string)
--- FAIL: TestEnvVarHandlerSpecialValue (0.00s)

FAIL
```


### How to Reproduce

Run the above test case.

### Expected Behavior

PASS

### Fiber Version

v2.37.0

### Code Snippet (optional)

_No response_

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

--- a/middleware/envvar/README.md
+++ b/middleware/envvar/README.md
@@ -44,10 +44,12 @@ app.Use("/expose/envvars", envvar.New())
 ### Custom Config
 
 ```go
-app.Use("/expose/envvars", envvar.New(envvar.Config{
-    ExportVars:  map[string]string{"testKey": "", "testDefaultKey": "testDefaultVal"},
-    ExcludeVars: map[string]string{"excludeKey": ""}}
-}))
+app.Use("/expose/envvars", envvar.New(
+	envvar.Config{
+		ExportVars:  map[string]string{"testKey": "", "testDefaultKey": "testDefaultVal"},
+		ExcludeVars: map[string]string{"excludeKey": ""},
+	}),
+)
 ```
 
 ### Response
@@ -57,10 +59,9 @@ Http response contract:
 {
   "vars": {
     "someEnvVariable": "someValue",
-    "anotherEnvVariable": "anotherValue",
+    "anotherEnvVariable": "anotherValue"
   }
 }
-
 ```
 
 ## Config
@@ -68,12 +69,11 @@ Http response contract:
 ```go
 // Config defines the config for middleware.
 type Config struct {
-    // ExportVars specifies the environment variables that should export
-    ExportVars map[string]string
-    // ExcludeVars specifies the environment variables that should not export
-    ExcludeVars map[string]string
+	// ExportVars specifies the environment variables that should export
+	ExportVars map[string]string
+	// ExcludeVars specifies the environment variables that should not export
+	ExcludeVars map[string]string
 }
-
 ```
 
 ## Default Config

--- a/middleware/envvar/envvar.go
+++ b/middleware/envvar/envvar.go
@@ -5,7 +5,6 @@ import (
 	"strings"
 
 	"github.com/gofiber/fiber/v2"
-	"github.com/gofiber/fiber/v2/utils"
 )
 
 // Config defines the config for middleware.
@@ -40,14 +39,10 @@ func New(config ...Config) fiber.Handler {
 		envVar := newEnvVar(cfg)
 		varsByte, err := c.App().Config().JSONEncoder(envVar)
 		if err != nil {
-			c.Response().SetBodyRaw(utils.UnsafeBytes(err.Error()))
-			c.Response().SetStatusCode(fiber.StatusInternalServerError)
-			return nil
+			return c.Status(fiber.StatusInternalServerError).SendString(err.Error())
 		}
-		c.Response().SetBodyRaw(varsByte)
-		c.Response().SetStatusCode(fiber.StatusOK)
-		c.Response().Header.Set("Content-Type", "application/json; charset=utf-8")
-		return nil
+		c.Set(fiber.HeaderContentType, fiber.MIMEApplicationJSONCharsetUTF8)
+		return c.Send(varsByte)
 	}
 }
 

--- a/middleware/envvar/envvar.go
+++ b/middleware/envvar/envvar.go
@@ -1,10 +1,11 @@
 package envvar
 
 import (
-	"github.com/gofiber/fiber/v2"
-	"github.com/gofiber/fiber/v2/utils"
 	"os"
 	"strings"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
 )
 
 // Config defines the config for middleware.
@@ -62,7 +63,7 @@ func newEnvVar(cfg Config) *EnvVar {
 		}
 	} else {
 		for _, envVal := range os.Environ() {
-			keyVal := strings.Split(envVal, "=")
+			keyVal := strings.SplitN(envVal, "=", 2)
 			if _, exists := cfg.ExcludeVars[keyVal[0]]; !exists {
 				vars.set(keyVal[0], keyVal[1])
 			}

--- a/middleware/envvar/envvar_test.go
+++ b/middleware/envvar/envvar_test.go
@@ -2,12 +2,13 @@ package envvar
 
 import (
 	"encoding/json"
-	"github.com/gofiber/fiber/v2"
-	"github.com/gofiber/fiber/v2/utils"
 	"io/ioutil"
 	"net/http"
 	"os"
 	"testing"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
 )
 
 func TestEnvVarStructWithExportVarsExcludeVars(t *testing.T) {
@@ -101,4 +102,39 @@ func TestEnvVarHandlerMethod(t *testing.T) {
 	resp, err := app.Test(req)
 	utils.AssertEqual(t, nil, err)
 	utils.AssertEqual(t, fiber.StatusMethodNotAllowed, resp.StatusCode)
+}
+
+func TestEnvVarHandlerSpecialValue(t *testing.T) {
+	testEnvKey := "testEnvKey"
+	fakeBase64 := "testBase64:TQ=="
+	os.Setenv(testEnvKey, fakeBase64)
+	defer os.Unsetenv(testEnvKey)
+
+	app := fiber.New()
+	app.Use("/envvars", New())
+	app.Use("/envvars/export", New(Config{ExportVars: map[string]string{testEnvKey: ""}}))
+
+	req, _ := http.NewRequest("GET", "http://localhost/envvars", nil)
+	resp, err := app.Test(req)
+	utils.AssertEqual(t, nil, err)
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+	utils.AssertEqual(t, nil, err)
+
+	var envVars EnvVar
+	utils.AssertEqual(t, nil, json.Unmarshal(respBody, &envVars))
+	val := envVars.Vars[testEnvKey]
+	utils.AssertEqual(t, fakeBase64, val)
+
+	req, _ = http.NewRequest("GET", "http://localhost/envvars/export", nil)
+	resp, err = app.Test(req)
+	utils.AssertEqual(t, nil, err)
+
+	respBody, err = ioutil.ReadAll(resp.Body)
+	utils.AssertEqual(t, nil, err)
+
+	var envVarsExport EnvVar
+	utils.AssertEqual(t, nil, json.Unmarshal(respBody, &envVarsExport))
+	val = envVarsExport.Vars[testEnvKey]
+	utils.AssertEqual(t, fakeBase64, val)
 }

